src/features/island/collectibles/components/GenieLamp.tsx

```diff
@@ -1,19 +1,132 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+
+import { Context } from "features/game/GameProvider";
 
 import genieLamp from "assets/sfts/genie_lamp.png";
 import { PIXEL_SCALE } from "features/game/lib/constants";
+import { Panel } from "components/ui/Panel";
+import { Revealing } from "features/game/components/Revealing";
+import { Revealed } from "features/game/components/Revealed";
+import { useActor } from "@xstate/react";
+import { Modal } from "react-bootstrap";
+import classNames from "classnames";
+
+import genieImg from "assets/npcs/genie.png";
+import { setImageWidth } from "lib/images";
+import { Button } from "components/ui/Button";
+import { CloseButtonPanel } from "features/game/components/CloseablePanel";
+
+interface Props {
+  id: string;
+}
+
+export const GenieLamp: React.FC<Props> = ({ id }) => {
+  const { gameService } = useContext(Context);
+  const [gameState] = useActor(gameService);
+
+  const lamps = gameState.context.state.collectibles["Genie Lamp"];
+  const lamp = lamps?.find((lamp) => lamp.id === id);
+  const rubbedCount = lamp?.rubbedCount ?? 0;
+  const wishesRemaining = 3 - rubbedCount;
+  const hasBeenRubbed = rubbedCount > 0;
+
+  const [isConfirming, setIsConfirming] = useState(false);
+  const [isRevealing, setIsRevealing] = useState(false);
+
+  // This useEffect is for the case where it is the last genie lamp wish.
+  // Genie Lamp will be deleted from the game state, so we need to make sure
+  // we do no get stuck in the revealing state.
+  useEffect(() => {
+    return () => {
+      gameService.send("CONTINUE");
+    };
+  }, []);
+
+  const rub = () => {
+    setIsConfirming(false);
+    setIsRevealing(true);
+
+    gameService.send("REVEAL", {
+      event: {
+        type: "genieLamp.rubbed",
+        id,
+        createdAt: new Date(),
+      },
+    });
+  };
 
-export const GenieLamp: React.FC = () => {
   return (
     <>
       <img
+        onClick={() => setIsConfirming(true)}
         src={genieLamp}
         style={{
           width: `${PIXEL_SCALE * 22}px`,
```

Comment:
This width doesn't seem to be applied because the container is 1 Grid Wide.

Here is an example of what the 22px would look like. I would suggest changing the dimensions to 1x2 and centering the lamp

<img width="121" alt="Screen Shot 2023-05-09 at 10 35 29 am" src="https://user-images.githubusercontent.com/11745561/236965168-3e197f62-7edc-4f54-bc37-956565646768.png">